math/src/fft/fft_inputs.rs

```diff
@@ -0,0 +1,103 @@
+use crate::FieldElement;
+use super::StarkField;
+use core::slice::ChunksMut;
+
+#[cfg(feature = "concurrent")]
+use utils::rayon::{self, slice::{Chunks, ChunksMut as ParChunksMut}};
+
+pub trait FftInputs<B: StarkField, E: FieldElement<BaseField = B>> {
+
+    fn evaluate(chunk: &mut Self, p: &Self, offset: B, factor: B, skip: usize);
+
+    fn interpolate(eval: &mut Self, domain_offset: B, offset: &mut B);
+
+    fn chunks_mut(&mut self, chunk_size: usize) -> ChunksMut<'_, E>;
```

Comment:
I was thinking of returning `Self` instead of a field element here- `ChunksMut<'_, Self>`. This way, we would get `chunks` of type `[Ffttrait]` in `evaluate_poly_with_offset`, which we can use in `fft_in_place`. While implementing this trait for `[E]`, the `chunks_mut` return a struct of type `ChunksMut<'_, E>` instead of `ChunksMut<'_, [E]>` due to which I had to introduce additionals generics to handle this.